src/main/scala/diplomaticobjectmodel/logicaltree/RocketLogicalTreeNode.scala

```diff
@@ -16,6 +16,11 @@ import freechips.rocketchip.tile.{RocketTileParams, TileParams, XLen}
  */
 class DCacheLogicalTreeNode(deviceOpt: Option[SimpleDevice], params: DCacheParams) extends LogicalTreeNode(() => deviceOpt) {
   def getOMComponents(resourceBindings: ResourceBindings, children: Seq[OMComponent]): Seq[OMComponent] = {
+    deviceOpt match {
+      case Some(device) => require(!resourceBindings.map.isEmpty, s"""ERROR: Memory region ${device.devname} is empty""")
+      case None =>
+    }
```

Comment:
Since you don't do anything with the `None` case, this is probably clearer as:

```scala
deviceOpt.forEach {
  device => require(!resourceBindings.map.isEmpty, s"""ERROR: Memory region ${device.devname} is empty""")
}
```

Also, the error message isn't correct. "device.devname" isn't a memory region; it's a Device. The real error is that if you have a device, then we expect to have resource bindings associated with the device to be passed in, not that the device doesn't have memory regions.

Reply:
fixed